dataland-frontend/tests/e2e/fixtures/eutaxonomy/non-financials/EuTaxonomyDataForNonFinancialsFixtures.ts

```diff
@@ -65,36 +60,31 @@ function generateActivity(): EuTaxonomyActivity {
   };
 }
 
-/**
- * Generates a map with keys drawn from the provided array and generated values
- * @param possibleKeys the keys that can occur in the resulting map
- * @param valueGenerator the generator function for the values
- * @returns the generated map
- */
-function generateMap<K, V>(possibleKeys: Array<K>, valueGenerator: () => V): Map<K, V> {
-  const keys = getRandomNumberOfDistinctElementsFromArray(Array.from(possibleKeys));
-  return new Map<K, V>(keys.map((key) => [key, valueGenerator()]));
-}
-
-/**
- * Generates an object with keys drawn from the provided objects values and generated values
- * @param possibleKeys the keys that can occur in the resulting map
- * @param valueGenerator the generator function for the values
- * @returns the generated map
- */
-function generateObject<V>(possibleKeys: Array<string>, valueGenerator: () => V): { [p: string]: V } {
-  return Object.fromEntries(generateMap(possibleKeys, () => valueGenerator()));
-}
-
 /**
  * Generates a random aligned activity
  * @returns a random aligned activity
  */
 function generateAlignedActivity(): EuTaxonomyAlignedActivity {
   return {
     ...generateActivity(),
-    substantialContributionCriteria: generateObject(Object.values(EnvironmentalObjective), generatePercentage),
-    dnshCriteria: generateObject(Object.values(EnvironmentalObjective), () =>
+    substantialContributionToClimateChangeMitigation: valueOrUndefined(generatePercentage()),
+    substantialContributionToClimateChangeAdaption: valueOrUndefined(generatePercentage()),
+    substantialContributionToSustainableUseAndProtectionOfWaterAndMarineResources: valueOrUndefined(
+      generatePercentage(),
+    ),
+    substantialContributionToTransitionToACircularEconomy: valueOrUndefined(generatePercentage()),
+    substantialContributionToPollutionPreventionAndControl: valueOrUndefined(generatePercentage()),
+    substantialContributionToProtectionAndRestorationOfBiodiversityAndEcosystems: valueOrUndefined(
+      generatePercentage(),
+    ),
+    dnshToClimateChangeMitigation: valueOrUndefined(faker.helpers.arrayElement(Object.values(YesNo))),
```

Comment:
This is personal preference, but as we use this a lot, I liked the `randomYesNo()` function more.

Reply:
Changed